File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.job import JobResponse


class ExtractedResultResponse(BaseModel):
    id: str
    document_id: str
    word_count: int | None = None
    readability_score: float | None = None
    primary_keywords: list[dict] | None = None
    auto_summary: str | None = None
    user_edited_summary: str | None = None
    is_finalized: bool = False
    finalized_at: datetime | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class DocumentResponse(BaseModel):
    id: str
    user_id: str
    original_filename: str
    file_size: int
    mime_type: str
    upload_timestamp: datetime
    created_at: datetime
    latest_job: JobResponse | None = None
    result: ExtractedResultResponse | None = None

    model_config = ConfigDict(from_attributes=True)


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int
    page: int
    page_size: int

File: frontend/src/types/document.ts
export interface JobResponse {
	id: string;
	document_id: string;
	status: string;
	progress_percentage: number;
	current_stage: string;
	error_message: string | null;
	retry_count: number;
	created_at: string;
	started_at: string | null;
	completed_at: string | null;
}

export interface DocumentResponse {
	id: string;
	user_id: string;
	original_filename: string;
	file_size: number;
	mime_type: string;
	upload_timestamp: string;
	created_at: string;
	latest_job?: JobResponse | null;
}

export interface KeywordMetric {
	keyword: string;
	count: number;
	density: number;
}

export interface ExtractedResultResponse {
	id: string;
	document_id: string;
	job_id: string;
	word_count: number | null;
	readability_score: number | null;
	primary_keywords: KeywordMetric[] | null;
	auto_summary: string | null;
	user_edited_summary: string | null;
	is_finalized: boolean;
	finalized_at: string | null;
	created_at: string;
}

export interface DocumentDetailResponse extends DocumentResponse {
	latest_job: JobResponse | null;
	result: ExtractedResultResponse | null;
}

export interface UploadResponse {
	document_id: string;
	job_id: string;
	filename: string;
	file_size: number;
}
